test/common.js

```diff
@@ -0,0 +1,3 @@
+const envPaths = require('env-paths')
+
+module.exports.devDir = () => envPaths('node-gyp', { suffix: '' }).cache
```

Comment:
Based on https://github.com/nodejs/build/issues/1908#issuecomment-544298898 this will make tests write into the user's home directory and not clean up afterwards.

Reply:
Yes, it does, the alternative is having a separate one that needs cleaning up afterward.

So test/test-configure-python.js doesn't need to test downloading, but downloading gets in the way, which is why we end up with directories in cwd when running the tests, so it needs to point to somewhere. If we point it to the user's node-gyp cache then any user likely running these tests is also fairly likely to already have the current version of Node's headers downloaded in that directory so it gets to skip the download part for the purpose of test/test-configure-python.js. But then since we're no longer including the download part in the tests (if the user has the headers), I introduced the new test/test-download.js which uses the same cache path but cleans it before running it then puts a downloaded version there and leaves it.

Messing with the user's home directory isn't ideal, but I also don't want to have multiple _actual_ downloads in a single test run, making the tests longer and also leaving more of a mess to clean up. At least this way the only mess we're making is either going to be a duplicate of what the user already has or will be useful for them when they next need the headers for the version they're running.

Taking too many liberties outside of cwd?

Reply:
> 
> Taking too many liberties outside of cwd?

Let's live with it for now.